editor/translations/editor_locale_dialog.cpp

```diff
@@ -31,546 +31,795 @@
+	vb->add_child(warn_lbl);
+}
 
-	TreeItem *c = cnt_list->get_selected();
-	if (c) {
-		country_code->set_text(c->get_metadata(0).operator String());
-	}
+/**************************************************************************/
+/* EditorLocaleDialog                                                     */
+/**************************************************************************/
+
+void EditorLocaleDialog::_bind_methods() {
+	ADD_SIGNAL(MethodInfo("locale_selected", PropertyInfo(Variant::STRING, "locale")));
 }
 
-void EditorLocaleDialog::_toggle_advanced(bool p_checked) {
-	if (!p_checked) {
-		script_code->set_text("");
-		variant_code->set_text("");
-	}
-	_update_tree();
+void EditorLocaleDialog::ok_pressed() {
+	emit_signal(SNAME("locale_selected"), locale.operator String());
+	hide();
 }
 
 void EditorLocaleDialog::_post_popup() {
 	ConfirmationDialog::_post_popup();
+}
 
-	if (!locale_set) {
-		lang_code->set_text("");
-		script_code->set_text("");
-		country_code->set_text("");
-		variant_code->set_text("");
-	}
-	edit_filters->set_pressed(false);
-	_update_tree();
+void EditorLocaleDialog::set_locale(const String &p_locale) {
+	locale_valid = !p_locale.is_empty();
+	locale = TranslationServer::Locale(p_locale, false);
+
+	_lang_search(lang_search->get_text());
+	_script_search(script_search->get_text());
+	_country_search(country_search->get_text());
+	variant_code->set_text(locale.variant);
+	_item_selected();
 }
 
-void EditorLocaleDialog::_filter_lang_option_changed() {
-	TreeItem *t = lang_list->get_edited();
-	String lang = t->get_metadata(0);
-	bool checked = t->is_checked(0);
+void EditorLocaleDialog::popup_locale_dialog() {
+	popup_centered_clamped(Size2(1050, 700) * EDSCALE, 0.8);
+}
 
-	Variant prev;
-	Array f_lang_all;
+void EditorLocaleDialog::_toggle_advanced(bool p_checked) {
+	script_vb->set_visible(p_checked);
+	variant_code->set_visible(p_checked);
+	variant_lbl->set_visible(p_checked);
+}
 
-	if (ProjectSettings::get_singleton()->has_setting("internationalization/locale/language_filter")) {
-		f_lang_all = GLOBAL_GET("internationalization/locale/language_filter");
-		prev = f_lang_all;
+void EditorLocaleDialog::_update_cache() {
+	if (updating_settings) {
+		return;
 	}
+	translation_cache.clear();
+}
 
-	int l_idx = f_lang_all.find(lang);
+void EditorLocaleDialog::_lang_search(const String &p_text) {
+	updating_lists = true;
+	lang_list->clear();
 
-	if (checked) {
-		if (l_idx == -1) {
-			f_lang_all.append(lang);
+	TreeItem *root = lang_list->create_item(nullptr);
+	lang_list->set_hide_root(true);
+	bool current_found = false;
+
+	// 1. Add search results.
+	TreeItem *search_root = lang_list->create_item(root);
+	search_root->set_icon(0, get_editor_theme_icon(SNAME("Search")));
+	search_root->set_selectable(0, false);
+	search_root->set_text(0, TTR("Search results"));
```

Comment:
```suggestion
	search_root->set_text(0, TTRC("Search results"));
```
Applies to all TTRs in this file.
Remember to override auto-translation mode for translated TreeItems.